libevmasm/Assembly.cpp

```diff
@@ -226,7 +226,7 @@ void Assembly::assemblyStream(ostream& _out, string const& _prefix, StringMap co
 		for (size_t i = 0; i < m_subs.size(); ++i)
 		{
 			_out << endl << _prefix << "sub_" << i << ": assembly {\n";
-			m_subs[i]->assemblyStream(_out, _prefix + "    ", _sourceCodes);
+			m_subs[i]->assemblyStream(_out, _prefix + "    ", std::move(_sourceCodes));
```

Comment:
This won't work.
You're inside a ``for`` loop - so if ``m_sub.size() > 1``, then you will have moved away ``_sourceCodes`` in the first iteration and move an empty string starting from the second iteration.